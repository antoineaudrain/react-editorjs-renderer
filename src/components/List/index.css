.er-list__ol {
    padding: .4em 0 .4em 40px;
    margin: 0;
    outline: none;
    list-style: decimal;
}

.er-list__ul {
    padding: .4em 0 .4em 40px;
    margin: 0;
    outline: none;
    list-style: disc;
}

.er-list__li {
    padding: 5.5px 0 5.5px 3px;
    line-height: 1.6em;
    font-size: 15px;
    color: #313649;
    letter-spacing: .005em;
}