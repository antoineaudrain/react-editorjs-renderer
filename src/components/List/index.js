import React from 'react'
import PropTypes from 'prop-types'
import ReactHtmlParser from 'react-html-parser'
import propsConstructor from '../../tools/propsConstructor'
import './index.css'

const List = ({ data, style = {} }) => {
  const getProps = propsConstructor(style)
  const orderedListStyle = getProps('list', 'ol')
  const unorderedListStyle = getProps('list', 'ul')
  const itemStyle = getProps('list', 'li')

  const content = data.items.map((item, index) => {
    const content = ReactHtmlParser(item)
    return (
      <li key={index} {...itemStyle}>
        {content}
      </li>
    )
  })

  return data.style === 'ordered' ? (
    <ol {...orderedListStyle}>{content}</ol>
  ) : (
    <ul {...unorderedListStyle}>{content}</ul>
  )
}

List.propTypes = {
  data: PropTypes.shape({
    style: PropTypes.oneOf(['ordered', 'unordered']).isRequired,
    items: PropTypes.array.isRequired
  }).isRequired,
  style: PropTypes.shape({
    ol: PropTypes.object,
    ul: PropTypes.object,
    li: PropTypes.object,
  })
}

export default List
