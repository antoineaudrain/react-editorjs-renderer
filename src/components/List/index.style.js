const style = {
  ol: {
    padding: '.4em 0 .4em 40px',
    margin: '0',
    outline: 'none',
    listStyle: 'decimal'
  },
  ul: {
    padding: '.4em 0 .4em 40px',
    margin: '0',
    outline: 'none',
    listStyle: 'disc'
  },
  li: {
    padding: '5.5px 0 5.5px 3px',
    lineHeight: '1.6em',
    fontSize: '15px',
    color: '#313649',
    letterSpacing: '.005em'
  }
}

export default style