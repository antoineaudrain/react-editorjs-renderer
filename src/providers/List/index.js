import React from 'react'
import PropTypes from 'prop-types'
import ReactHtmlParser from 'react-html-parser'
import providerPropsConstructor from '../../tools/providerPropsConstructor'
import defaultStyle from './index.style'

const List = ({ data, style = {} }) => {
  const getProps = providerPropsConstructor(defaultStyle, style)
  const orderedListStyle = getProps('ordered-list')
  const unorderedListStyle = getProps('unordered-list')
  const itemStyle = getProps('item')

  const content = data.items.map((item, index) => {
    const content = ReactHtmlParser(item)
    return (
      <li key={index} {...itemStyle}>
        {content}
      </li>
    )
  })

  return data.style === 'ordered' ? (
    <ol {...orderedListStyle}>{content}</ol>
  ) : (
    <ul {...unorderedListStyle}>{content}</ul>
  )
}

List.propTypes = {
  data: PropTypes.shape({
    style: PropTypes.oneOf(['ordered', 'unordered']).isRequired,
    items: PropTypes.array.isRequired
  }).isRequired,
  style: PropTypes.object
}

export default List
