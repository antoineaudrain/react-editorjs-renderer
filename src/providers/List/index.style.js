const style = {
  'ordered-list': {
    padding: '.4em 0',
    margin: 0,
    paddingLeft: 40,
    outline: 'none',
    listStyle: 'disc'
  },
  'unordered-list': {
    padding: '.4em 0',
    margin: 0,
    paddingLeft: 40,
    outline: 'none',
    listStyle: 'decimal'
  },
  items: {
    padding: '5.5px 0 5.5px 3px',
    lineHeight: '1.6em'
  }
}

export default style
