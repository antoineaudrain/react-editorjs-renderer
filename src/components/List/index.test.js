import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import List from './index'

configure({ adapter: new Adapter() })

describe('List provider:', () => {
  const orderedList = shallow(<List data={{ style: 'ordered', items: [] }} />)
  const unorderedList = shallow(
    <List data={{ style: 'unordered', items: ['test'] }} />
  )

  it('should match snapshot', () => expect(orderedList).toMatchSnapshot())
  it('should match snapshot', () => expect(unorderedList).toMatchSnapshot())
})
