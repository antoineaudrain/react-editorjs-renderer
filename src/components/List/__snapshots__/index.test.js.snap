// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List provider: should match snapshot 1`] = `ShallowWrapper {}`;

exports[`List provider: should match snapshot 2`] = `ShallowWrapper {}`;
